nautobot/docs/user-guide/administration/installation/health-checks.md

```diff
@@ -0,0 +1,307 @@
+
+```yaml
+livenessProbe:
+  exec:
+    command:
+      - "/bin/bash"
+      - "-c"
+      - "[ $(find $NAUTOBOT_CELERY_BEAT_HEARTBEAT_FILE -mmin -0.1 | wc -l) -eq 1 ] || false"
+  initialDelaySeconds: 30
+  periodSeconds: 5
+  timeoutSeconds: 5
+  failureThreshold: 3
+```
+
+### PostgreSQL Container in k8s
+
+```yaml
+readinessProbe:
+  exec:
+    command:
+      - "/bin/bash"
+      - "-c"
+      - "pg_isready -d $POSTGRES_DB -U $POSTGRES_USER"
+  initialDelaySeconds: 30
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 3
+
+livenessProbe:
+  exec:
+    command:
+      - "/bin/bash"
+      - "-c"
+      - "pg_isready -d $POSTGRES_DB -U $POSTGRES_USER"
+  initialDelaySeconds: 30
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 3
+```
+
+### MySQL Container in k8s
+
+```yaml
+readinessProbe:
+  exec:
+    command:
+      - "/bin/bash"
+      - "-c"
+      - 'mysql -u $MYSQL_USER --password=$MYSQL_PASSWORD -h localhost --execute "SHOW DATABASES;"'
+  initialDelaySeconds: 30
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 3
+
+livenessProbe:
+  exec:
+    command:
+      - "/bin/bash"
+      - "-c"
+      - "mysqladmin -u $MYSQL_USER --password=$MYSQL_PASSWORD -h localhost ping"
+  initialDelaySeconds: 30
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 3
+```
+
+### Redis Container in k8s
+
+```yaml
+readinessProbe:
+  exec:
+    command:
+      - "/bin/bash"
+      - "-c"
+      - "redis-cli -h localhost ping | grep PONG"
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 3
+
+livenessProbe:
+  exec:
+    command:
+      - "/bin/bash"
+      - "-c"
+      - "redis-cli -h localhost ping"
+  periodSeconds: 10
+  timeoutSeconds: 5
+  failureThreshold: 3
+```
+
+## Docker Compose Deployments
+
+Docker Compose supports a single `healthcheck` for each container.
+
+### Nautobot Server Container in Docker Compose
+
+We recommend the CLI-based health-check rather than the HTTP health-check here because the former will not fail when all request-handler workers are busy.
+
+```yaml
+healthcheck:
+  interval: 10s
+  timeout: 10s
+  start_period: 5m
+  retries: 3
+  test:
+    - "CMD"
+    - "nautobot-server"
+    - "health_check"
+```
+
+### Celery Worker Container in Docker Compose
+
+```yaml
+healthcheck:
+  interval: 60s
+  timeout: 10s
+  start_period: 30s
+  retries: 3
+  test:
+    - "CMD"
+    - "/bin/bash"
+    - "-c"
+    - "nautobot-server celery inspect ping --destination celery@$$HOSTNAME"
+```
+
+### Celery Beat Container in Docker Compose
+
+```yaml
+healthcheck:
+  interval: 5s
+  timeout: 5s
+  start_period: 30s
+  retries: 3
+  test:
+    - "CMD"
+    - "/bin/bash"
+    - "-c"
+    - "[ $$(find /tmp/nautobot_celery_beat_heartbeat -mmin -0.1 | wc -l) -eq 1 ] || false"
+```
+
+### PostgreSQL Container in Docker Compose
+
+```yaml
+healthcheck:
+  interval: 10s
+  timeout: 5s
+  start_period: 30s
+  retries: 3
+  test:
+    - "CMD-SHELL"
```

Comment:
Any reason we're using `CMD-SHELL` here but explicitly invoking bash in the other examples?

Reply:
Fixed in next commit - simplified the `tests` for docker-compose and updated our docker-compose.yml example accordingly.